src/start/registers.md

```diff
@@ -15,19 +18,30 @@ You may well find that the code you need to access the peripherals in your micro
 [embedded-hal]: https://crates.io/crates/embedded-hal
 [Portability]: ../portability/index.md
 [F3]: https://crates.io/crates/f3
+[Discovery]: https://rust-embedded.github.io/discovery/
 
+## Board Crate
 
-## Starting at the bottom
+A board crate is the perfect starting point, if you're new to embedded Rust. They nicely abstracts the HW details that might be overwelming when starting studying this subject, and makes standard tasks easy, like turning a LED on or off. The functionality they exposes varies a lot between boards. Since this book aims at staying hardware agnostic, the board crates won't be covered by this book.
+
+If you want to experiment with the STM32F3DISCOVERY board, it is highly recommmand to take a look at the [F3] board crate, which provides functionality to blink the board LEDs, access its compass, bluetooth and more. The [Discovery] book offers a great introduction to the [F3] board crate.
+
+But if you're working on a system that doesn't yet have dedicated board crate, or you need functionality not provided by existing crates, read on as we start from the bottom, with the micro-architecture crates.
+
+## Micro-architecture crate
 
 Let's look at the SysTick peripheral that's common to all Cortex-M based micro-controllers. We can find a pretty low-level API in the [cortex-m] crate, and we can use it like this:
 
 ```rust,ignore
+#![no_std]
+#![no_main]
 use cortex_m::peripheral::{syst, Peripherals};
 use cortex_m_rt::entry;
+extern crate panic_halt;
```

Comment:
```suggestion
use panic_halt as _;
```

Reply:
I've updated all occurrences on the page.